mlflow/projects/databricks.py

```diff
@@ -106,24 +217,41 @@ def _create_databricks_run(tracking_uri, experiment_id, source_name, source_vers
                                     source_type=SourceType.PROJECT)
 
 
+def _parse_dbfs_uri_path(dbfs_uri):
+    """Parses and returns the absolute path within DBFS of the file with the specified URI"""
+    return urllib.parse.urlparse(dbfs_uri).path
+
+
 def run_databricks(uri, entry_point, version, parameters, experiment_id, cluster_spec,
                    git_username, git_password):
     """
     Runs a project on Databricks, returning a `SubmittedRun` that can be used to query the run's
     status or wait for the resulting Databricks Job run to terminate.
     """
-    # Create run object with remote tracking server
+    _check_databricks_cli_configured()
+    if cluster_spec is None:
+        raise ExecutionException("Cluster spec must be provided when launching MLflow project runs "
+                                 "on Databricks.")
+
+    # Fetch the project into work_dir & validate parameters
+    work_dir = _get_work_dir(uri, use_temp_cwd=False)
+    _fetch_project(uri, version, work_dir, git_username, git_password)
+    project = _load_project(work_dir, uri)
+    project.get_entry_point(entry_point)._validate_parameters(parameters)
+    # Upload the project to DBFS, get the URI of the project
+    final_experiment_id = experiment_id or Experiment.DEFAULT_EXPERIMENT_ID
+    dbfs_project_uri = _upload_project_to_dbfs(work_dir, final_experiment_id)
+
+    # Create run object with remote tracking server. Get the git commit from the working directory,
+    # etc.
     tracking_uri = tracking.get_tracking_uri()
     remote_run = _create_databricks_run(
-        tracking_uri=tracking_uri, experiment_id=experiment_id, source_name=uri,
-        source_version=version, entry_point_name=entry_point)
+        tracking_uri=tracking_uri, experiment_id=experiment_id, source_name=project.uri,
+        source_version=tracking._get_git_commit(work_dir), entry_point_name=entry_point)
     # Set up environment variables for remote execution
-    env_vars = {"MLFLOW_GIT_URI": uri}
```

Comment:
+1 to removing these. Should we also remove `git_username` and `git_password` from all argument lists, and `mlflow run` command line arguments?